### PR TITLE
Revert 'Revert "TRT-1588: Avoid NULL for prowjob_start_ts"'

### PR DESCRIPTION
**Pull request by DennisPeriquet**
Reverts https://github.com/openshift/sippy/pull/1584 and puts my fix on top.

/hold

My fix will surely fix the error we saw when the original PR was reverted.  But I'd like to test this locally before merging to ensure the other changes don't break a `fetchdata` operation using a test method that runs an actual `fetchdata` operation that uses bigquery (e.g., like the sippy backup script).

### CODE REVIEW

**Comment by DennisPeriquet** [comment]
trying locally using a similar script as the sippy backup script and see these errors after [here](https://github.com/openshift/sippy/blob/5cb32239ad424222f3e2019a617a4acc315875d7/pkg/dataloader/prowloader/bigqueryjobs.go#L75):

```
ERRO[2024-04-03T21:46:08.451-04:00] error parsing prowjob from bigquery           error="bigquery: schema field prowjob_start_ts of type TIMESTAMP is not assignable to struct field prowjob_start_ts of type *time.Time"
ERRO[2024-04-03T21:46:08.451-04:00] error parsing prowjob from bigquery           error="bigquery: schema field prowjob_start_ts of type TIMESTAMP is not assignable to struct field prowjob_start_ts of type *time.Time"
ERRO[2024-04-03T21:46:08.451-04:00] error parsing prowjob from bigquery           error="bigquery: schema field prowjob_start_ts of type TIMESTAMP is not assignable to struct field prowjob_start_ts of type *time.Time"
ERRO[2024-04-03T21:46:08.451-04:00] error parsing prowjob from bigquery           error="bigquery: schema field prowjob_start_ts of type TIMESTAMP is not assignable to struct field prowjob_start_ts of type *time.Time"
```

**Comment by DennisPeriquet** [comment]
the fetchdata is working using similar command line as the sippy backup script; will let it continue ...

**Comment by DennisPeriquet** [comment]
fetchdata finsihed with `no errors encountered during db refresh`.

/test e2e

**Comment by stbenjam** [comment]
> trying locally using a similar script as the sippy backup script and see these errors after [here](https://github.com/openshift/sippy/blob/5cb32239ad424222f3e2019a617a4acc315875d7/pkg/dataloader/prowloader/bigqueryjobs.go#L75):
> 
> ```
> ERRO[2024-04-03T21:46:08.451-04:00] error parsing prowjob from bigquery           error="bigquery: schema field prowjob_start_ts of type TIMESTAMP is not assignable to struct field prowjob_start_ts of type *time.Time"
> ERRO[2024-04-03T21:46:08.451-04:00] error parsing prowjob from bigquery           error="bigquery: schema field prowjob_start_ts of type TIMESTAMP is not assignable to struct field prowjob_start_ts of type *time.Time"
> ERRO[2024-04-03T21:46:08.451-04:00] error parsing prowjob from bigquery           error="bigquery: schema field prowjob_start_ts of type TIMESTAMP is not assignable to struct field prowjob_start_ts of type *time.Time"
> ERRO[2024-04-03T21:46:08.451-04:00] error parsing prowjob from bigquery           error="bigquery: schema field prowjob_start_ts of type TIMESTAMP is not assignable to struct field prowjob_start_ts of type *time.Time"
> ```

You want to use `bigquery.NullableTimestamp`

**Comment by DennisPeriquet** [comment]
fetchdata test got `no errors encountered during db refresh`:

```
./sippy load \
  --init-database \
  --load-openshift-ci-bigquery \
  --mode=ocp \
  --config=./config/openshift.yaml \
  --release=4.16 \
  --arch=amd64 \
  --database-dsn=postgresql://postgres:<redacted>@localhost:5482/postgres \
  --google-service-account-credential-file ...openshift-gce-devel-sippy.json
```

File: pkg/apis/prow/prow.go [hunk 1]
```diff
@@ -67,7 +71,7 @@ type ProwJobSpec struct {
 }
 
 type ProwJobStatus struct {
-	StartTime        time.Time               `json:"startTime,omitempty"`
+	StartTime        bigquery.NullTimestamp  `json:"startTime,omitempty"`
```
Comment (neisw) [inline pkg/apis/prow/prow.go:74]: I don't think you want to change this.  Keep it as time.Time

Check IsValid for bigqueryProwJobRun StartTime before it gets converted to ProwJobStatus and if it isn't valid log an error and skip creating that job, go on to the next one.

 https://github.com/openshift/sippy/blob/c32e1fcf0bfab7b2c7254a468509d6c1aa464c8a/pkg/dataloader/prowloader/bigqueryjobs.go#L99

Comment (DennisPeriquet) [reply pkg/apis/prow/prow.go]: nice suggestion Thanks.

File: pkg/dataloader/prowloader/bigqueryjobs.go [hunk 2]
```diff
@@ -52,6 +52,7 @@ func (pl *ProwLoader) fetchProwJobsFromOpenShiftBigQuery() ([]prow.ProwJob, []er
 		"FROM `ci_analysis_us.jobs` " +
 		`WHERE TIMESTAMP(prowjob_completion) > @queryFrom
 	       AND prowjob_url IS NOT NULL
+	       AND prowjob_start IS NOT NULL
```
Comment (neisw) [inline pkg/dataloader/prowloader/bigqueryjobs.go:55]: Technically we shouldn't need this any longer correct?

Comment (DennisPeriquet) [reply pkg/dataloader/prowloader/bigqueryjobs.go]: correct, since we're now accounting for it later.

File: pkg/apis/prow/prow.go [hunk 3]
```diff
@@ -1,6 +1,8 @@
 package prow
 
-import "time"
+import (
+	"time"
+)
```
Comment (neisw) [inline pkg/apis/prow/prow.go:5]: So if we are still blocked by the e2e test, this is really a noop and we could drop this file from the changelist.  Didn't notice it earlier and wouldn't hold this up if everything passed.

Comment (DennisPeriquet) [reply pkg/apis/prow/prow.go]: I pulled it out of the changed list.  Interesting how vscode kept the parens.

**Pull request merged**